pkg/controller/controller_instance.go

```diff
@@ -254,8 +281,8 @@ func (c *controller) reconcileServiceInstanceAdd(instance *v1beta1.ServiceInstan
 		return nil
 	}
 
-	if instance.Status.ReconciledGeneration == instance.Generation {
-		glog.V(4).Info(pcb.Message("Not processing event because reconciled generation showed there is no work to do"))
+	if isServiceInstanceProcessedAlready(instance) {
```

Comment:
I'm not sure that this is going to hold as we stop doing multiple status updates in the control-loops.  For example, if we refactor the control loop so that:

1. In the first iteration of the `ServiceInstance` control loop, we update the references and stop the loop
2. In the second iteration, we update the status to show that we're starting provision, and exit the loop
3. In the third iteration, we actually make the provision call, and update the status according to the result of that call

Based my current reading, it looks like this PR and that refactor would not play nicely together.

Reply:
It would play fine if we set the `Ready:False` condition as part of step 1.

Reply:
As @staebler pointed out, we just don't set `Ready:True` until we have finished the processing of a particular generation (i.e. until iteration 3 or even 4 has finished either with success or a non-recoverable error like 400).